### PR TITLE
CI: fail if duplicates found

### PR DESCRIPTION
**Pull request by tjhei**
based on conversation in https://github.com/dealii/publication-list/pull/385#issuecomment-1235675330

### CODE REVIEW

**Comment by tjhei** [comment]
works!
```
+ grep Duplicate
WARN - Duplicate entry key: 'Jodlbauer2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Guo2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Deolmi2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Cheng2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Cangiani2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Zhang2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Carreo2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Wang2018' in file '../publications-2018.bib', skipping ...
WARN - Duplicate entry key: 'Arbogast2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Guo2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Hagstrom2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Balicki2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'EbnaHai2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Bzowski2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Freddi2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Alarydah2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Carreo2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Vassaux2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Dang2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Heltai2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Kim2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Bonito2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Carreo2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Vassaux2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Konschin2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Robey2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Das2019' in file '../publications-2019.bib', skipping ...
WARN - Duplicate entry key: 'Zhang2019' in file '../publications-2019.bib', skipping ...
+ exit 1
```

waiting to rebase to fix these...

**Review by marcfehling — APPROVED** [review]
Nice!

File: Jenkinsfile [hunk 1]
```diff
@@ -68,7 +68,8 @@ pipeline
             sh '''
                cd offline
                pdflatex -interaction=nonstopmode publication_list.tex
-               biber publication_list
+               biber publication_list | tee biber.log
+               grep WARN biber.log | grep Duplicate && exit 1
```
Comment (marcfehling) [inline Jenkinsfile:72]:
```suggestion
               biber publication_list
               grep WARN publication_list.blg | grep Duplicate && exit 1
```

`biber` already writes a logfile, which you could use alternatively. I leave the choice to you :)

```
$ grep WARN publication_list.blg | grep Duplicate
[18283] Biber.pm:130> WARN - Duplicate entry key: 'Jodlbauer2018' in file '../publications-2018.bib', skipping ...
[18283] Biber.pm:130> WARN - Duplicate entry key: 'Guo2018' in file '../publications-2018.bib', skipping ...
[18283] Biber.pm:130> WARN - Duplicate entry key: 'Deolmi2018' in file '../publications-2018.bib', skipping ...
[18283] Biber.pm:130> WARN - Duplicate entry key: 'Cheng2018' in file '../publications-2018.bib', skipping ...
[18283] Biber.pm:130> WARN - Duplicate entry key: 'Cangiani2018' in file '../publications-2018.bib', skipping ...
[18283] Biber.pm:130> WARN - Duplicate entry key: 'Zhang2018' in file '../publications-2018.bib', skipping ...
[18284] Biber.pm:130> WARN - Duplicate entry key: 'Carreo2018' in file '../publications-2018.bib', skipping ...
[18284] Biber.pm:130> WARN - Duplicate entry key: 'Wang2018' in file '../publications-2018.bib', skipping ...
[18285] Biber.pm:130> WARN - Duplicate entry key: 'Arbogast2019' in file '../publications-2019.bib', skipping ...
[18285] Biber.pm:130> WARN - Duplicate entry key: 'Guo2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Hagstrom2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Balicki2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'EbnaHai2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Bzowski2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Freddi2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Alarydah2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Carreo2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Vassaux2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Dang2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Heltai2019' in file '../publications-2019.bib', skipping ...
[18286] Biber.pm:130> WARN - Duplicate entry key: 'Kim2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Bonito2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Carreo2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Vassaux2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Konschin2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Robey2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Das2019' in file '../publications-2019.bib', skipping ...
[18287] Biber.pm:130> WARN - Duplicate entry key: 'Zhang2019' in file '../publications-2019.bib', skipping ...
```

**Comment by bangerth** [comment]
Can you try to rebase and see whether the warnings are now gone?

**Comment by marcfehling** [comment]
We still need to merge the 2019 duplicate patch #385.

**Comment by tjhei** [comment]
done.

**Comment by marcfehling** [comment]
We currently build a bibliography for every year. I found the entries in #390 with `biber` in 'tool' mode when creating the union of all bibliographies.

I guess it is more important that we can build the offline document without `biber` skipping duplicates, so let's merge this!

**Comment by bangerth** [comment]
I'll let @tjhei do the merge.

Separately, I think that the de-duplication has added 30-40 entries to the total number on the mailing list. I did not investigate, but the number jumped from ~1810 to ~1850 now and I don't think we added that many recently. I can only imagine that jabref silently dropped a bunch of entries. This suggests that a CI check is indeed useful.

**Comment by marcfehling** [comment]
> Separately, I think that the de-duplication has added 30-40 entries to the total number on the mailing list. I did not investigate, but the number jumped from ~1810 to ~1850 now and I don't think we added that many recently. I can only imagine that jabref silently dropped a bunch of entries. This suggests that a CI check is indeed useful.

We just recently started counting the 2022 publications with #383.

**Comment by bangerth** [comment]
Oh yes, that would have done it!

**Pull request merged**